api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -980,3 +984,22 @@ type Taint struct {
 	// Valid effects are NoSchedule, PreferNoSchedule and NoExecute.
 	Effect corev1.TaintEffect `json:"effect"`
 }
+
+// Diagnostics specifies the diagnostics settings for a virtual machine.
+// +kubebuilder:validation:XValidation:rule="self.storageAccountType == 'UserManaged' ? has(self.storageAccountURI) : true", message="storageAccountURI is required when storageAccountType is UserManaged"
+type Diagnostics struct {
+	// StorageAccountType determines if the storage account for storing the diagnostics data
+	// should be disabled (Disabled), provisioned by Azure (Managed) or by the user (UserManaged).
+	// +kubebuilder:validation:Enum=Managed;UserManaged;Disabled
+	StorageAccountType string `json:"storageAccountType,omitempty"`
```

Comment:
Should there be any default value?

Reply:
I'd prefer it be explicit but I wouldn't hold up the PR over it. My reasoning is the same as https://github.com/openshift/hypershift/pull/3970/files#r1587489752.